tests/migrations/test_autodetector.py

```diff
@@ -2251,6 +2251,33 @@ def test_fk_dependency_other_app(self):
         self.assertOperationAttributes(changes, 'testapp', 0, 0, name="book")
         self.assertMigrationDependencies(changes, 'testapp', 0, [("otherapp", "__first__")])
 
+    def test_alter_field_to_fk_dependency_other_app(self):
+        """
+        #30691 - Alter field to foreign key should create dependency on other apps.
+        """
+        before = [
+            ModelState('testapp', 'EntityA', [
+                ('id', models.AutoField(primary_key=True)),
+            ]),
+            ModelState('otherapp', 'EntityB', [
+                ('id', models.AutoField(primary_key=True)),
+                ('entity_a', models.UUIDField())
```

Comment:
Might want to make this an `IntegerField` to avoid taking alteration to an incompatible type into account (e.g. uuid to int).